pipeline/matching/rankers/quantity.py

```diff
@@ -6,12 +6,30 @@
 RANKER_QUANTITY = "quantityRanker"
 
 
+def get_number_matches(matches):
+    num_matches = {}
+    for match in matches:
+        for user in match.users:
+            if num_matches.get(user) is None:
+                num_matches[user] = 1
+            else:
+                num_matches[user] += 1
+    return num_matches
+
+
 class QuantityRanker(MatchRanker):
     def __init__(self):
         super().__init__(name=RANKER_QUANTITY)
 
     def rank(
         self, inp: MatchingInput, proposed: Iterator[Match]
     ) -> Iterator[Match]:
-        # TODO: Implement your ranker
-        yield from proposed
+        """The function 'rank' returns a sorted iterator based on the total number of matches for each user.
+        The less matches a user has the higher they are prioritized."""
+        num_matches = get_number_matches(proposed)
+        yield from sorted(
+            proposed,
+            key=lambda match: sum(
+                [num_matches.get(user) for user in match.users]
+            ),
```

Comment:
This is cool! At first, I was thinking we would rank matches by the user in that match with the fewest number of options, not by the sum of the number of matches for all the users in the match.

I realized that your sum approach will work as well, so let's stick with this!

In the future, when we combine this with other rankers, it might be better to use the minimum number of options rather than the sum of options, that way other rankers can have an effect on which match to choose for a user with few options, rather than just matching them to the user with the fewest options. But we can investigate this later when there are more generators and rankers in production.

Reply:
Sticking with current implementation, keeping this approach in mind if a user with a low number of matches is getting low quality matches.